output/createMaxCSV.sh

```diff
@@ -19,115 +19,241 @@
+if [[ ! -f $GRIDFILE && ! -f $INPUTDIR/$GRIDFILE && ! -f "fort.14" ]]; then
+   echo "ERROR: Mesh file '$GRIDFILE' specified in run.properties file was not found in $INPUTDIR or '.'; nor was 'fort.14'." | tee $SCENARIOLOG
+   exit 1
+fi
+fort14=$GRIDFILE
+if [[ ! -f $GRIDFILE ]]; then
+   if [[ -f $INPUTDIR/$GRIDFILE ]]; then
+      fort14=$INPUTDIR/$GRIDFILE
+   else
+      fort14="fort.14"
+   fi
+fi
+awk 'NR==2 { np=$2 } NR>2 && NR<=np+2 { print $2","$3","$4 }' $fort14 > xyd.txt 2>> $SCENARIOLOG
+#
+# create metadata for summary column definitions and column units and
 # convert the netcdf files to ascii and then extract just the data values
 # from the resulting ascii files
 columnDefinitions="# longitude,latitude,depth,maxele"
 columnUnits="# degrees east,degrees north,m below datum,m above datum"
 windFile=wind10m.maxwvel.txt
 waveFile=swan_HS_max.txt
-${OUTPUTDIR}/netcdf2adcirc.x --datafile maxele.63.nc 2>> $SYSLOG
+netcdf2adcirc.x --datafile maxele.63.nc 2>> $SCENARIOLOG
 awk 'NR>3 { print $2 }' maxele.63 > maxele.txt
-if [[ -e wind10m.maxwvel.nc  ]]; then
+if [[ -e wind10m.maxwvel.63.nc  ]]; then
    columnDefinitions="${columnDefinitions},wind10m.maxwvel"
-   columnUnits="${columnUnits},m/s at 10m above ground" 
-   ${OUTPUTDIR}/netcdf2adcirc.x --datafile wind10m.maxwvel.63.nc 2>> $SYSLOG
+   columnUnits="${columnUnits},m/s at 10m above ground"
+   netcdf2adcirc.x --datafile wind10m.maxwvel.63.nc 2>> $SYSLOG
    mv maxwvel.63 wind10m.maxwvel.63 2>> $SYSLOG
    awk 'NR>3 { print $2 }' wind10m.maxwvel.63 > wind10m.maxwvel.txt
-else 
-   columnDefinitions="${columnDefinitions},maxwvel" 
-   columnUnits="${columnUnits},m/s at ground level" 
-   ${OUTPUTDIR}/netcdf2adcirc.x --datafile maxwvel.63.nc 2>> $SYSLOG
+elif [[ -e maxwvel.63.nc  ]]; then
+   columnDefinitions="${columnDefinitions},maxwvel"
+   columnUnits="${columnUnits},m/s at ground level"
+   netcdf2adcirc.x --datafile maxwvel.63.nc 2>> $SYSLOG
    awk 'NR>3 { print $2 }' maxwvel.63 > maxwvel.txt
    windFile=maxwvel.txt
+else
+   echo "WARNING: Wind file not found."
+   unset windFile
 fi
-if [[ -e swan_HS_max.nc  ]]; then
+if [[ -e swan_HS_max.63.nc  ]]; then
    columnDefinitions="${columnDefinitions},swan_HS_max"
-   columnUnits="${columnUnits},m above msl" 
-   ${OUTPUTDIR}/netcdf2adcirc.x --datafile swan_HS_max.63.nc 2>> $SYSLOG
+   columnUnits="${columnUnits},m above msl"
+   netcdf2adcirc.x --datafile swan_HS_max.63.nc 2>> $SYSLOG
    awk 'NR>3 { print $2 }' swan_HS_max.63 > swan_HS_max.txt
 else
-   waveFile=''
+   echo "WARNING: Wave file not found."
+   unset waveFile
 fi
 # place all columns in one file with a comma as the delimiter
 paste -d "," xyd.txt maxele.txt $windFile $waveFile > max_data.csv
 # add header to data
 echo $columnDefinitions >> header.csv
 echo $columnUnits >> header.csv
 cat header.csv max_data.csv > $csvFileName 2>> $SYSLOG
-gzip $csvFileName 2>> $SYSLOG
+gzip -f $csvFileName 2>> $SYSLOG &
```

Comment:
Can you explain the purpose of this change (`-f`) and why it is being backgrounded? Unless we don't care if this is successful or not, then we should avoid backgrounding processes with out a `wait` and check on the status - if the purpose is for parallelizing process efficiency; I think we may still need to check the status of the process when it is done.

Reply:
The `-f` is to relieve me of the need to make sure I have deleted the the `*.gz` files from a previous execution of this script before executing it again. There are dozens of files to compress so it takes a long time if I do them 1 at a time (and my desktop CPU has 16 cores). Yes, I agree that the script is way too complicated. :-)

Reply:
Hm....we should look at using `pigz` - I am pretty sure I can add this easily to the install.